### PR TITLE
MM-10514: Remove delete channel link for town-square in classic app

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Remove delete channel link for town-square in mobile classic apps.

#### Ticket Link
[MM-10514](https://mattermost.atlassian.net/browse/MM-10514)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: components/navbar/navbar.jsx [hunk 1]
```diff
@@ -647,7 +647,9 @@ export default class Navbar extends React.Component {
                             </li>
                         </ChannelPermissionGate>
                     );
+                }
 
+                if (!ChannelStore.isDefault(channel)) {
```
Comment (saturninoabril) [inline components/navbar/navbar.jsx]: Can we add test snapshots of `createDropdown` with non-default and default channels?

Comment (jespino) [reply components/navbar/navbar.jsx]: Sure :)

Comment (jespino) [reply components/navbar/navbar.jsx]: Done!

**Comment by jespino** [comment]
Oops. Fixing :)

**Pull request merged**